test/scheduler_hooks_test.rb

```diff
@@ -21,4 +21,36 @@
     assert_equal(0, Resque.delayed_timestamp_size(enqueue_time.to_i),
                  'job should not be enqueued')
   end
+
+  test 'default failure hooks are called when enqueueing a job fails' do
+    config = {
+      'cron' => '* * * * *',
+      'class' => 'SomeRealClass',
+      'args' => '/tmp'
+    }
+
+    e = RuntimeError.new('custom error')
+    Resque::Scheduler.stubs(:enqueue_from_config).raises(e)
+
+    Resque::Scheduler.expects(:on_failure_default).with(config, e)
+    Resque::Scheduler.enqueue(config)
+  end
+
+  test 'failure hooks are called when enqueueing a job fails' do
+    Resque::Scheduler.failure_handler = ExceptionHandlerClass
```

Comment:
this is going to leak into other tests... should unset it in an `ensure` block to the value it had before.. maybe make a `with_failure_handler` test helper for that?